frontend/components/groups/js/groups.service.ts

```diff
@@ -0,0 +1,502 @@
+		for (const key in group.models) {
+
+			if (!group.models.hasOwnProperty(key)) {
+				continue;
+			}
+
+			const meshIds = group.models[key].meshes;
+			const pair = key.split("@");
+			const account = pair[0];
+			const model = pair[1];
+
+			this.ViewerService.resetMeshColor(
+				account,
+				model,
+				meshIds,
+				group.color,
+			);
+		}
+
+		delete this.state.colorOveride[groupId];
+
+	}
+
+	/**
+	 * Reselect a group
+	 */
+	public reselectGroup(group) {
+		this.selectGroup(group);
+	}
+
+	/**
+	 * Return all the meshes for all currently highlighted objects
+	 */
+	public getCurrentMeshHighlights() {
+
+		return this.TreeService.getCurrentMeshHighlights().then((objects) => {
+
+			let total = 0;
+			for (const key in objects) {
+				if (objects[key] && objects[key].meshes) {
+					total += objects[key].meshes.length;
+				}
+			}
+			return total;
+		});
+	}
+
+	/**
+	 * Return a default group name, incrementing upwards
+	 */
+	public getDefaultGroupName(groups) {
+		const groupNames = [];
+		groups.forEach((group) => {
+			groupNames.push(group.name);
+		});
+
+		const prefix = "Group ";
+		let num = 1;
+		let groupName = prefix + num;
+		while (groupNames.indexOf(groupName) !== -1) {
+			groupName = prefix + num++;
+		}
+		return groupName;
+	}
+
+	/**
+	 * Set the color of the selected group to a given color
+	 */
+	public setSelectedGroupColor(color) {
+		this.state.selectedGroup.color = color;
+		this.updateSelectedGroupColor();
+	}
+
+	/**
+	 * Return an CSS friendly RGBA color from an array
+	 */
+	public getRGBA(color) {
+		const red = color[0];
+		const blue = color[1];
+		const green = color[2];
+		return `rgba(${red}, ${blue}, ${green}, 1)`;
+	}
+
+	/**
+	 * Return an CSS friendly RGBA for a given group
+	 */
+	public getGroupRGBAColor(group) {
+		if (group && group.color) {
+			return this.getRGBA(group.color);
+		}
+
+		return "rgba(255, 255, 255, 1)";
+	}
+
+	/**
+	 * Return a random RGB color array i.e. [255, 255, 255]
+	 */
+	public getRandomColor(): number[] {
+		return [
+			parseInt((Math.random() * 255).toFixed(0), 10),
+			parseInt((Math.random() * 255).toFixed(0), 10),
+			parseInt((Math.random() * 255).toFixed(0), 10),
+		];
+	}
+
+	/**
+	 * Convert a colour from hex to an RGBA value
+	 */
+	public hexToRGBA(hex: string, alpha: number) {
+
+		alpha = (alpha !== undefined) ? alpha : 1;
+
+		// Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
+		const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i;
+		hex = hex.replace(shorthandRegex, function(m, r, g, b) {
+			return r + r + g + g + b + b;
+		});
+
+		const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+		return result ? [
+			parseInt(result[1], 16),
+			parseInt(result[2], 16),
+			parseInt(result[3], 16),
+		] : [];
+
+	}
+
+	/**
+	 * Do any cleaning that's necessary for groups in the UI
+	 */
+	public cleanGroups(groups: any[]) {
+		groups.forEach((group) => {
+			if (!group.name) {
+				group.name = "No assigned name";
```

Comment:
put it in brackets so it's clear(er) that someone didn't just name it as "No assigned name".

I would probably just write `(No name)`, but up to you